### PR TITLE
[MODULAR] Adds beefmen

### PR DESCRIPTION
**Pull request by linnpap**
## About The Pull Request
Ports the Beefmen species from Fulpstation, with a few nerfs to make them more conducive to RP.

image

Beefmen are fast, but fragile. They also have unique bleeding mechanics.

However, the ability for Beefmen to teleport, their limb swap ability, and their brain traumas have been removed, in order to promote better RP. Along with this, there is a disclaimer that role playing standards still apply when you spawn as a Beefman.

## Why It's Good For The Game
More species are always fun, and I'm sure quite a few players would like to play as Beefmen. However, I am aware there is going to be a full-scale nuclear war over whether or not Beefmen are LRP or not, due to their association with Fulp. I will just say this: we already have some other ridiculous species (ethereals, plasmamen, LITERAL SKELETONS), so I really doubt Beefmen are that crazy, at least in SS13.

## Changelog
:cl:
add: Added Beefmen
add: Added sashes for Beefmen
/:cl:

### CODE REVIEW

**Comment by jinxynii** [comment]
if u add my squidward in too ill give u a thumbs up...

**Comment by Trubbl3** [comment]
https://www.youtube.com/watch?v=tr3RUHuS_Lk

**Comment by JohnFulpWillard** [comment]
> However, I am aware there is going to be a full-scale nuclear war over whether or not Beefmen are LRP or not, due to their association with Fulp.

😕 Don't really see the need to insult our server when you're trying to port our code over.
Also, are you not porting the sashes too?

**Comment by linnpap** [comment]
> 
> 
> > However, I am aware there is going to be a full-scale nuclear war over whether or not Beefmen are LRP or not, due to their association with Fulp.
> 
> 😕 Don't really see the need to insult our server when you're trying to port our code over.

please i didnt mean it like that PLEASE IM SORRY

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/beefman.dm [hunk 1]
```diff
@@ -0,0 +1,415 @@
...
+						I.reagents.trans_to(user, I.reagents.total_volume)	//transfer reagents to player
+					qdel(I)
+					playsound(get_turf(H), 'sound/effects/meatslap.ogg', 50, 1)
+
+			return TRUE // True CANCELS the sequence.
+
+	return ..() // TRUE FALSE
+
+/datum/species/beefman/after_equip_job(datum/job/J, mob/living/carbon/human/H, visualsOnly = FALSE)
+	to_chat(H, "<span class='danger'>Just because meatmen are from Fulpstation, which is more loose rule-wise than Skyrat, DOES NOT MEAN YOU ARE MAGICALLY ALLOWED TO BE LRP. ROLEPLAYING STANDARDS STILL APPLY.</span>")
```
Comment (Draganfrukts) [inline modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/beefman.dm]:
```suggestion
	to_chat(H, "<span class='danger'>Meatmen come from Fulpstation, a server with a looser ruleset than ours. This is NOT a pass to grief. Policy still applies!</span>")
```
I think this wording would be more appropiate

**Comment by Nobelium-XVIII** [comment]
> > However, I am aware there is going to be a full-scale nuclear war over whether or not Beefmen are LRP or not, due to their association with Fulp.
> 
> 😕 Don't really see the need to insult our server when you're trying to port our code over.
> Also, are you not porting the sashes too?

they didn't even merge the updated version...

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/beefman.dm (hunk 1 above)
Comment (SgtHunk) [reply modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/beefman.dm]: Also why isn't it using the span macros?

**Comment by FlamingLily** [comment]
> > However, I am aware there is going to be a full-scale nuclear war over whether or not Beefmen are LRP or not, due to their association with Fulp.
> 
> 😕 Don't really see the need to insult our server when you're trying to port our code over.

I don't think they intended it that way, more "I know people on this server will cause shit because of associations with fulp, but personally I'm not going to feed into that"

**Comment by Nobelium-XVIII** [comment]
please fix the checks

File: modular_skyrat/modules/customization/modules/clothing/under/sashes.dm [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+/obj/item/clothing/under/bodysash/civilian
+	name = "civilian's sash"
+	icon_state = "civilian"
+
+///Misc
+
+/obj/item/clothing/under/bodysash/russia
+	name = "Russian sash"
+	icon_state = "russia"
```
Comment (SgtHunk) [inline modular_skyrat/modules/customization/modules/clothing/under/sashes.dm:210]: This sash isn't used anywhere unless you plan on porting the ruin for it.

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/beefman.dm [hunk 3]
```diff
@@ -0,0 +1,519 @@
...
+		if("Warden")
+			newSash = new /obj/item/clothing/under/bodysash/warden()
+		if("Security Officer")
+			newSash = new /obj/item/clothing/under/bodysash/security()
+		if("Detective")
+			newSash = new /obj/item/clothing/under/bodysash/detective()
+		if("Brig Physician")
+			newSash = new /obj/item/clothing/under/bodysash/brigdoc()
+		if("Deputy")
+			newSash = new /obj/item/clothing/under/bodysash/deputy()
```
Comment (SgtHunk) [inline modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/beefman.dm]: are you sure you have these jobs in the first place?

Comment (thestubborn) [reply modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/beefman.dm]: we have the former but not the latter

File: modular_skyrat/modules/customization/modules/clothing/under/sashes.dm (hunk 2 above)
Comment (linnpap) [reply modular_skyrat/modules/customization/modules/clothing/under/sashes.dm:210]: may as well still have the item

**Comment by ZephyrTFA** [comment]
Gonna be honest here. This entire species seems like a bad joke at best and a meme at worst. Reading your code and the comments within solidifies my opinion aswell. Comments such as "BLEED YOUR JUICES" and the interesting sounds you've chosen to override the defaults with.

**Comment by linnpap** [comment]
if anyone has any ideas to make beefmen perceived as less memey PLEASE FOR THE LOVE OF GOD give me suggestions
i really want them in skyrat from the bottom of my heart

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/beefman.dm [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+
+	skinned_type = /obj/item/food/meatball
+	meat = /obj/item/food/meat/slab //What the species drops on gibbing
+	toxic_food = DAIRY | PINEAPPLE
+	disliked_food = VEGETABLES | FRUIT | CLOTH
+	liked_food = RAW | MEAT
+	attack_verb = "meat"
+	payday_modifier = 0.75 //-- "Equality"
+	speedmod = -0.2 // this affects the race's speed. positive numbers make it move slower, negative numbers make it move faster
+	armor = -2 // overall defense for the race... or less defense, if it's negative.
```
Comment (SgtHunk) [inline modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/beefman.dm]: what's the point of them taking 2% more damage?

File: modular_skyrat/modules/customization/modules/clothing/under/sashes.dm [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+	name = "brig physician's sash"
+	icon_state = "brigdoc"
+	sensor_mode = SENSOR_COORDS
+	random_sensor = FALSE
+
+/obj/item/clothing/under/bodysash/deputy
+	name = "deputy's sash"
+	icon_state = "deputy"
+	sensor_mode = SENSOR_COORDS
+	random_sensor = FALSE
```
Comment (SgtHunk) [inline modular_skyrat/modules/customization/modules/clothing/under/sashes.dm]: What's the point of keeping a sash for a job you don't have? Also, shouldn't you add more sashes for skyrat-exclusive jobs?

**Comment by Gandalf2k15** [comment]
+1 from me.

**Comment by Gandalf2k15** [comment]
Suggestion: allow them to reattach limbs but have guns absolutely obliterate them. So like imagine chunks of meat flying off. You can achieve this by making limbs fly off if they're shot.

**Comment by lyricalpaws** [comment]
(This is mostly opinion) I'd enjoy the idea of beefmen a lot more if they were reflavoured in to some sort of ghoulish species. Rather than their entire thing being "They're walking liver paté". A civilised ghoul (ala fallout) would probably go down with the community a lot better. People could still make them in to a beefperson, but it wouldn't be the default, if that makes sense.

**Comment by tf-4** [comment]
-1

Meme race with no redeeming qualities.

**Comment by hugmeorelse** [comment]
@linnpap Can we make it less meme-y? Less meme name. Make it feel roleplay-oriented rather than a big meme or joke race? It needs to be reworked to be serious.

**Comment by hugmeorelse** [comment]
> (This is mostly opinion) I'd enjoy the idea of beefmen a lot more if they were reflavoured in to some sort of ghoulish species. Rather than their entire thing being "They're walking liver paté". A civilised ghoul (ala fallout) would probably go down with the community a lot better. People could still make them in to a beefperson, but it wouldn't be the default, if that makes sense.

Doesn't have to be this, but something like this in a sense. It just needs to be reworked to be a more serious race.

Anyways, if we make it less meme-y and more serious, I think having them be like an event race to start with would be good.

**Comment by hugmeorelse** [comment]
Hm, why not reframe them as flesh golems or something like that?

**Comment by Horatio22** [comment]
I'm not really comfortable with this port, but since I resprited most of the job sashes, I did want to point out that they do not cover the groin at all. This was done intentionally to show off the beefman's unique body sprites (and they canonically don't have a sex or gender so this wasn't an issue), but I don't know how that will interact with Skyrat's anatomy customization. Seeing as any humanoid can wear a sash, this may be something to consider with the clothing part of this PR at least.

image

**Comment by linnpap** [comment]
funding has been dropped for the beefmen
It's over.

**Comment by hugmeorelse** [comment]
:(

**Pull request closed without merging**